### PR TITLE
Fix unit selection of CupyMemoryProfiler

### PR DESCRIPTION
**Pull request by kmaehashi**
Currently output unit of CupyMemoryProfiler always becomes bytes.
This issue is introduced in #6256.

### CODE REVIEW

File: chainer/function_hooks/cupy_memory_profile.py [hunk 1]
```diff
@@ -129,10 +129,12 @@ def summary(self):
     def _choose_unit(self, size):
         """Choose optimal unit."""
         denomi = 1
+        if size <= 0:
+            return denomi, self._units[0]
```
Comment (toslunar) [inline chainer/function_hooks/cupy_memory_profile.py]: :+1:

File: chainer/function_hooks/cupy_memory_profile.py [hunk 2]
```diff
@@ -129,10 +129,12 @@ def summary(self):
...
         for unit in self._units[:-1]:
-            if size / denomi >= 1:
+            if size / (denomi * 1024.0) < 1:
```
Comment (toslunar) [inline chainer/function_hooks/cupy_memory_profile.py]: How about fixing the previous line?
`for unit in self._units:`

File: chainer/function_hooks/cupy_memory_profile.py [hunk 3]
```diff
@@ -129,10 +129,12 @@ def summary(self):
...
                 return denomi, unit
-            denomi *= 1024.0
+            denomi *= 1024
```
Comment (toslunar) [inline chainer/function_hooks/cupy_memory_profile.py]: It seems `from __future__ import division` will be needed (L170--172).

File: chainer/function_hooks/cupy_memory_profile.py (hunk 2 above)
Comment (kmaehashi) [reply chainer/function_hooks/cupy_memory_profile.py]: Sorry, I couldn't get your point. Will that simplify this code?

File: chainer/function_hooks/cupy_memory_profile.py (hunk 3 above)
Comment (kmaehashi) [reply chainer/function_hooks/cupy_memory_profile.py]: Thanks, changed to always return denomi as float, as it was the original spec.

File: chainer/function_hooks/cupy_memory_profile.py (hunk 2 above)
Comment (toslunar) [reply chainer/function_hooks/cupy_memory_profile.py]: I'm sorry.  I was confused with `self._units[::-1]`.

**Comment by toslunar** [comment]
CIs, test this please.

**Comment by kmaehashi** [comment]
The error is not related to this PR.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**